pkg/controller/combo_controller.go

```diff
@@ -2,36 +2,161 @@ package controller
 
 import (
 	"context"
+	"fmt"
+	"strings"
+	"time"
 
 	"github.com/go-logr/logr"
+	"github.com/operator-framework/combo/api/v1alpha1"
+	combinationPkg "github.com/operator-framework/combo/pkg/combination"
+	templatePkg "github.com/operator-framework/combo/pkg/template"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/errors"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
-
-	"github.com/operator-framework/combo/api/v1alpha1"
+	"sigs.k8s.io/controller-runtime/pkg/source"
 )
 
 type combinationController struct {
 	client.Client
 	log logr.Logger
 }
 
+// manageWith creates a new instance of this controller
 func (c *combinationController) manageWith(mgr ctrl.Manager, version int) error {
 	c.log = c.log.V(version)
+	templateHandler := handler.EnqueueRequestsFromMapFunc(c.mapTemplateToCombinations)
+
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&v1alpha1.Combination{}).
+		Watches(&source.Kind{Type: &v1alpha1.Template{}}, templateHandler).
 		Complete(c)
 }
 
+// mapTemplateToCombinations is responsible for taking the template object and finding all associated
+// combinations that should be requeued. This should only happen whenever a template is changed in someway.
+func (c *combinationController) mapTemplateToCombinations(template client.Object) []reconcile.Request {
+	if template == nil {
+		return nil
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	templateName := template.GetName()
+
+	// Find all of the combinations that rely on this template and enqueue them for updates
+	var requests []reconcile.Request
+	combinationList := v1alpha1.CombinationList{}
+	if err := c.List(ctx, &combinationList, &client.ListOptions{Namespace: template.GetNamespace()}); err != nil {
```

Comment:
Templates are [cluster scoped](https://github.com/operator-framework/combo/blob/main/manifests/crds/combo.io_templates.yaml#L17) so the namespace will always be `""`. [Combinations are also cluster scoped](https://github.com/operator-framework/combo/blob/main/manifests/crds/combo.io_combinations.yaml#L17), so this is effectively "Get every combination", so there is not a need to provide a list option.

Alternatively, you can have the combination controller apply a label to the `combination CR` that has some key (example: `combo.ReferencedTemplate: <templateNameDefinedInSpec>`) which could decrease the number of combinations that you itterrate over.